rpc/src/main/scala/WorkerServer.scala

```diff
@@ -61,8 +65,20 @@ class WorkerServer(executionContext: ExecutionContext) { self =>
       val promise = Promise[SampleReply]
 
       Future {
-        // TODO: Logic
-        promise.success(new SampleReply())
+        try {
+          val sortedBlock = Block.fromPath(Path("data/block"), 10, 90).sort()
+          val sampledKeys = sortedBlock
+            .sample()
+            .map(key => ByteString.copyFrom(key.underlying))
+            .toList
+          val reply = SampleReply(sampledKeys)
+
+          promise.success(reply)
+        } catch {
+          case e: Exception =>
+            println(e)
+            promise.failure(e)
+        }
```

Comment:
There is no useage of numberOfKeys which is in the SampleRequest

Reply:
Is there any purpose of `numberOfKey`? As I didn't understand its purpose, I haven't used it.